components/browser/search/src/main/java/mozilla/components/browser/search/SearchEngineManager.kt

```diff
@@ -99,6 +99,15 @@ class SearchEngineManager(
         }
     }
 
+    /**
+     * Returns the provided default search engine if the user has no preference.
+     */
+    @Synchronized
+    fun getProvidedDefaultSearchEngine(context: Context): SearchEngine {
+        val searchEngineList = getSearchEngineList(context)
+        return searchEngineList.default ?: searchEngineList.list[0]
+    }
```

Comment:
nit: We could use that method from `getDefaultSearchEngine()` now to avoid duplicating this logic.

Reply:
If we use getDefaultSearchEngine() in this function, we would have to provide a name that is not empty and doesn't match any of the search engine such as ".". Should we continue with such a change? 

Alternatively, we could call `getProvidedDefaultSearchEngine` in the providedDefault in `getDefaultSearchEngine` on line 94 to avoid some duplication in logic. What do you think?

Reply:
I meant the reverse: `getDefaultSearchEngine()` should call your new `getProvidedDefaultSearchEngine()` method for the default instead of duplicating this part.